### DISCUSSION TITLE
Audio Channel Counts and Formats

### DISCUSSION THREAD

**Discussion by jtc-dolby** [vsf-tv/gccg-api#36] [General]
I see that the main profile in the TR-11 draft imposes a limit of 16 audio channels. Why impose this restriction?
The topic of audio channel counts is often discussed in SVG and with our broadcast partners implementing next generation audio such as NBC, France TV and Canal+ etc. The general consensus in the audio community is that the 16 channel limit inherited from SDI is very restrictive and prohibits many of the new use cases being proposed. Audio in the future will be more immersive and have individual elements broken out to allow personalization. Very little personalization and immersive audio can be combined without breaking the 16 channel limit. I would propose a limit of 64 channels for a flow which is in line with ST 2110 and MADI. This is closely related to the need for ST 2110-41 metadata support to describe the audio essence bundle.
In [connection_schema.json](https://github.com/vsf-tv/gccg-api/blob/main/connection_schema.json) the limit is specified as 32 channels whereas in [payload_schema.json](https://github.com/vsf-tv/gccg-api/blob/main/payload_schema.json) the limit is 24. The latter number seems like it should be referring to bit depth. Either way the documents appear inconsistent.

Another issue related to this is the reuse of channel orderings from ST 2110-30 in connection_schema.json. This do not support immersive audio formats that are commonly used today for live sports. The addition of a few immersive formats (5.1.2, 5.1.4, 7.1.2, 7.1.4, 9.1.4, 9.1.6) would help. Ideally there would be a provision to describe the audio layout with ADM (ITU-R BS. 2076). This would support separated music and effects, dialogue, audio description etc. which is in the early stages of adoption.

The DCI profile allows 32 channels but I don't see any provision for object based audio essence (OBAE) as referenced in the DCI spec (3.3.1) and specified in ST 2098. This format is commonly used for studio theatrical releases.

I agree with a previous comment that a floating point format would be more machine native and should be considered for C-C. Fixed point is more of a limitation imposed by transmission. Single point floating point should be sufficient for all current applications although perhaps a provision for double precision floating point should be considered.
James Cowdery
Dolby Laboratories

**Comment by david3a** [comment]
If we allow multi-channel audio, we should certainly have no constraints that we will likely hit for many years! 64 seems a sensible starting point, I'll update our upcoming changes to this, any higher bidders?

**Reply by jtc-dolby** [reply]
The main use-case to go beyond 64 channels would be object-based audio. For example, a Dolby Atmos Master file (BWAV-ADM, MXF) today requires 128 channels and is widely used in theatrical and music production. However, with object-based audio you need to know where the objects are located etc. and that requires metadata, probably in S-ADM format. This relates to my previous suggestion to support ST 2110-41 metadata (S-ADM) as well as ST291 ANC. If support for object audio is added, to avoid hitting constraints for many years, I would suggest 256.
James Cowdery
Dolby Laboratories

**Reply by david3a** [reply]
and in practice this is just setting the upper limit so no resources are consumed unless we use the channels.

**Comment by david3a** [comment]
My other thought here is that why would we have an audio component with 0 channels? Should we set the minimum to 1?

**Reply by jtc-dolby** [reply]
Yes. Minimum of 1.
